patches/server/0965-Make-banners-modifiable-with-both-BannerMeta-and-Blo.patch

```diff
@@ -0,0 +1,274 @@
+         final int original;
+         int hash = original = super.applyHash();
++        if (true) return original; // Paper - CraftMetaBlockState's hash is good enough
+         if (this.base != null) {
+             hash = 31 * hash + this.base.hashCode();
+         }
+@@ -217,6 +281,7 @@ public class CraftMetaBanner extends CraftMetaItem implements BannerMeta {
+ 
+     @Override
+     public boolean equalsCommon(CraftMetaItem meta) {
++        if (true) return super.equalsCommon(meta); // Paper - Only check CraftMetaBlockState's data
+         if (!super.equalsCommon(meta)) {
+             return false;
+         }
+@@ -230,11 +295,13 @@ public class CraftMetaBanner extends CraftMetaItem implements BannerMeta {
+ 
+     @Override
+     boolean notUncommon(CraftMetaItem meta) {
++        if (true) return super.notUncommon(meta);  // Paper - Only check CraftMetaBlockState's data
+         return super.notUncommon(meta) && (meta instanceof CraftMetaBanner || (this.patterns.isEmpty() && this.base == null));
+     }
+ 
+     @Override
+     boolean isEmpty() {
++        if (true) return super.isEmpty(); // Paper - Only check CraftMetaBlockState's data
+         return super.isEmpty() && this.patterns.isEmpty() && this.base == null;
+     }
+ 
+@@ -246,7 +313,7 @@ public class CraftMetaBanner extends CraftMetaItem implements BannerMeta {
+     @Override
+     public CraftMetaBanner clone() {
+         CraftMetaBanner meta = (CraftMetaBanner) super.clone();
+-        meta.patterns = new ArrayList<>(this.patterns);
++        // meta.patterns = new ArrayList<>(this.patterns); // Paper - Only copy CraftMetaBlockState's data
+         return meta;
+     }
+ }
+diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBlockState.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBlockState.java
+index 1957e82b67b4aa301d12468d5a81d96e621dab49..0fdb24d6716de62e6e01174a10648875825cb437 100644
+--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBlockState.java
++++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBlockState.java
+@@ -40,7 +40,7 @@ public class CraftMetaBlockState extends CraftMetaItem implements BlockStateMeta
+             Material.BLACK_SHULKER_BOX
+     );
+ 
+-    private static final Set<Material> BLOCK_STATE_MATERIALS = Sets.newHashSet(
++    static final Set<Material> BLOCK_STATE_MATERIALS = Sets.newHashSet( // Paper - Change it to package-private, so we can use that in CraftMetaBanner
+             Material.FURNACE,
+             Material.CHEST,
+             Material.TRAPPED_CHEST,
+@@ -130,7 +130,7 @@ public class CraftMetaBlockState extends CraftMetaItem implements BlockStateMeta
+     @ItemMetaKey.Specific(ItemMetaKey.Specific.To.NBT)
+     static final ItemMetaKey BLOCK_ENTITY_TAG = new ItemMetaKey("BlockEntityTag");
+ 
+-    final Material material;
++    Material material; // Paper - Remove final keyword, so it can be changed in CraftMetaBanner's deserialized constructor
```

Comment:
Shouldn't be needed, generally this shouldn't change so a final should be kept here. See other comment for workaround.